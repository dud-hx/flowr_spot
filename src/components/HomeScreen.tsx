import { inject, observer } from 'mobx-react'
import React, { useEffect } from 'react'
import StateStore from '../state/stateStore'
import FlowerList from './flowerComponents/FlowerList'
import SearchFlowers from '../components/flowerComponents/SearchFlowers'
 interface IHomeProps {
  StateStore?: StateStore
}
const HomeScreen: React.FC<IHomeProps> = props => {
  const { StateStore } = props
  useEffect(() => {
    StateStore?.fetchFlowerData();
  }, [])
 
  return (
    <>
      <div className="home_header">
        <h1>Discover flowers around you</h1>
        <p>Explore between more than 8.427 sightings</p>
        <div style={{display: 'flex', justifyContent:'center' }}> <SearchFlowers /></div>
      </div>

      <FlowerList />   </>
  )
}

export default inject("StateStore")(observer(HomeScreen))