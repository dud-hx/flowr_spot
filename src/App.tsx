import React from "react";
import { Routes, BrowserRouter, Route } from "react-router-dom";
import "./assets/style/index.scss";
import Login from "./components/authenticationComponents/Login";
import Signup from "./components/authenticationComponents/Signup";
import FavoriteFlowers from "./components/flowerComponents/FavoriteFlowers";
import FlowerList from "./components/flowerComponents/FlowerList";
import HomeScreen from "./components/HomeScreen";
import Navigation from "./components/navigation/Navigation";
import SightingList from "./components/sightingsComponents/SightingList";

const App: React.FC<{}> = () => {
  return (
    <div className="app">
      <Navigation />
         <Routes>
          <Route path='/' element={<HomeScreen />} />
          <Route path="favorites" element={<FavoriteFlowers />}/>
          <Route path="flowers" element={<FlowerList />}/>
          <Route path="login" element={<Login />} />
          <Route path="signup" element={<Signup />} />
          <Route path="sighting" element={<SightingList />} />
        </Routes>
       
    </div>
  );
};

export default App;
