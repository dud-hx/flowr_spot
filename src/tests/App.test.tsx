import { render, screen } from '@testing-library/react';
import { Provider } from 'mobx-react';
import { BrowserRouter, Link } from 'react-router-dom';
import App from '../App';
import HomeScreen from '../components/HomeScreen';
import StateStore from '../state/stateStore';

 describe("<App />", () => {
  test("Should contain a logo with  name logo.svg", async () => {
    render(<BrowserRouter >
      <App /></BrowserRouter>)
        const data = screen.getByTestId('logo-id');
        expect(data.getAttribute('src')).toEqual('logo.svg');

  });
});

const ReactTestRenderer = require('react-test-renderer');

it('renders correctly', () => {
  const tree = ReactTestRenderer.create(<Provider StateStore={StateStore}><HomeScreen /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
