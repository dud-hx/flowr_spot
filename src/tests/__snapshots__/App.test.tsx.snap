// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="home_header"
  >
    <h1>
      Discover flowers around you
    </h1>
    <p>
      Explore between more than 8.427 sightings
    </p>
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
       
      <div
        className="MuiFormControl-root css-qvhgdm-MuiFormControl-root"
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorError MuiInputBase-formControl MuiInputBase-adornedEnd search__bar css-9xw4gu-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="search-bar"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Looking for something specific"
            required={false}
            type="text"
            value=""
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="toggle password visibility"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-gmng1k-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-ihdtdm"
            >
              <span
                className="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-13mmjd8-MuiGrid-root"
  />,
  "   ",
]
`;
